core/iwasm/include/wasm_export.h

```diff
@@ -729,6 +739,11 @@ wasm_runtime_validate_native_addr(wasm_module_inst_t module_inst,
 /**
  * Convert app address(relative address) to native address(absolute address)
  *
+ * Note that all native addresses to module instance memory can be
+ * invalidated on a memory growth. Thus, this API is basically unsafe
+ * to use for module instances with multiple threads unless you somehow
```

Comment:
It should be safe when multi-thread mode is enabled in WAMR, in this mode the linear memory is shared and is allocated with its maximum size, and won't be re-allocated again, we just increase the memory's current page count when memory growing:

https://github.com/WebAssembly/threads/blob/main/proposals/threads/Overview.md#shared-linear-memory

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_runtime.c#L2161

Reply:
WebAssembly's `thread` feature depends on the `shared_memory` feature, because the threads should have same "heap". If a wasm module's memory is not marked as `shared`, then it is not possible to create threads for it.

Reply:
ok. thank you for explanation. i will update.

Reply:
done